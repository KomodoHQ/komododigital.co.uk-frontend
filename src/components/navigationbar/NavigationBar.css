@import '../settings.css';

.navigationBar {
    position: fixed;
    padding: 0;
    color: #ffffff;
    transition: all 0.25s linear;
    z-index: 999;
}

.navigationBar > a {
    grid-column: 2 / 10;
    display: flex;
    align-items: center;
    text-decoration: none;
}

.navigationBar, .navigationBar a { color: white; }
.navigationBar.inverted, .navigationBar.inverted a { color: black; }

/* .navigationBar img {
    height: auto; 
    width: auto; 
    max-width: 235px;
    max-height: 235px
    transition: all 0.25s linear;
} */

.navigationBar > .hamburger {
    grid-column: 10 / 17;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-end;
    z-index: 99999;
}

.navigationBar > .hamburger > a {
    text-decoration: none;
    font-size: 24px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-end;

}

.navigationBar > ul {
    padding: 0;
    margin: 0;
    position: fixed;
    top: 0;
    left: -120vw;
    width: 100vw;
    height: 100vh;
    list-style: none;
    flex-direction: column;
    align-items: center;
    justify-content: flex-end;
    transition: all 0.25s linear;
}

.navigationBar > ul.open {
    transform: translate(120vw);
}

.navigationBar > ul > li {
    padding: 10px 50px;
}

.navigationBar > ul > li > a {
    text-decoration: none;
    line-height: 60px;
    font-size: 24px;
    transition: all 0.25s linear;
}

.navigationBar.small {
    padding: 10px 0;
    -webkit-box-shadow: 0px 5px 4px rgba(0,0,0,0.5);
    -moz-box-shadow: 0px 5px 4px rgba(0,0,0,0.5);
    box-shadow: 0px 5px 4px rgba(0,0,0,0.5);
}

/* .navigationBar.small img {
    max-width: 235px; 
    max-height: 235px;
} */

.navigationBar.small > ul > li > a {
    line-height: 45px;
    font-size: 20px;
}

@media (--mobile-viewport) {

    .navigationBar {
        padding: 10px 0;
    }

    .navigationBar > .hamburger {
        display: none;
    }

    .navigationBar > a {
        grid-column: 2 / 7;
    }

    .navigationBar > ul {
        list-style: none;
        grid-column: 7 / 16;
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        position: initial;
        transform: none;
        left: initial;
        height: auto;
        width: 100%;
    }

    .navigationBar > ul > li {
        padding: 0;
    }
    
    .navigationBar > ul > li > a {
        text-decoration: none;
        font-size: 16px;
    }
/* 
    .navigationBar.small img {
        max-width: 135px; 
        max-height: 135px;
    } */

}

@media (--laptop-viewport) {

    .navigationBar > ul > li > a {
        font-size: 24px;
    }

}

@media (--desktop-viewport) {

    .navigationBar > h1 {
        grid-column: 3 / 6;
    }

    .navigationBar > ul {
        grid-column: 9 / 16;
    }
}
