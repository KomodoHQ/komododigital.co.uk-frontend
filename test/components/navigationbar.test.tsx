import React from 'react';
import NavigationBar from '../../src/components/navigationbar';
import renderer from 'react-test-renderer';

test('NavigationBar element is rendered', () => {
  const data = {
  };
  const component = renderer.create(<NavigationBar data={data} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
