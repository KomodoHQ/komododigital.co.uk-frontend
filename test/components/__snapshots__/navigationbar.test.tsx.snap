// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar element is rendered 1`] = `
<div
  className="navigationBar"
>
  <a
    aria-current="page"
    href="/"
    onClick={[Function]}
  >
    <h1>
      Komodo
    </h1>
  </a>
  <ul>
    <li>
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        About
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Case Studies
      </a>
    </li>
    <li>
      <a
        href="/blog-list"
        onClick={[Function]}
      >
        Insights
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Contact
      </a>
    </li>
  </ul>
</div>
`;
